### PR TITLE
Allow sublayer/variable reassignment after build()

### LINKED ISSUE #18601: Consider allowing sublayer reassignment
We are having a lot of discussion around lora/quantization and other PEFT strategies that require replacing layers (often dense layers), with parameter efficient replacements.

In torch, a `nn.Module` will track submodules by name. So you can run `module.sub_module = new_sub_module` without issue. The old child will be booted out for the new. Same for `tf.Module`.

This is not the case with Keras layers. Tracking is currently "append only," not by attr name, and locked after `build()`. To work around this for a LoRA implementation, we currently do the following.

```python
def replace(parent, name, replacement):
    locked = parent._tracker.locked
    parent._tracker.locked = False
    target = getattr(parent, name)
    setattr(parent, name, replacement)
    parent._layers[parent._layers.index(target)] = replacement
    parent._tracker.locked = locked
```

We should consider whether we want to allow this for Keras layers with public APIs. Potentially by extending the tracker to track by attr name.

### PR DESCRIPTION
**Pull request by timon0305**
This change enables reassigning tracked attributes (sublayers, variables, metrics, seed generators) after a layer has been built. Previously, trying to reassign these attributes after build() would fail because the tracker was locked.

The fix modifies `Layer.__setattr__` to detect when an already-tracked attribute is being reassigned to a new value of the same type, and uses the existing `Tracker.replace_tracked_value()` method to perform the swap instead of failing.

This is useful for fine-tuning workflows where users may want to replace a sublayer (e.g., swap a classifier head) after the model has been built.

Fixes #18601

### CODE REVIEW

File: keras/src/layers/layer.py [hunk 1]
```diff
@@ -1581,6 +1581,22 @@ def __setattr__(self, name, value):
...
+                        is_attr_type(old_value)
+                        and is_attr_type(value)
+                        and self._tracker.is_in_store(store_name, old_value)
+                    ):
+                        self._tracker.replace_tracked_value(
+                            store_name, old_value, value
+                        )
+                        # Skip normal tracking since we've handled it
+                        super().__setattr__(name, value)
+                        return
```
[the remark this answers is not in the document]
Comment (jeffcarp) [reply keras/src/layers/layer.py:1599]: Please fix
